src/web/middleware/precondition.rs

```diff
@@ -3,20 +3,19 @@ use std::{cell::RefCell, rc::Rc};
 
 use crate::web::middleware::sentry::queue_report;
 use crate::web::{
-    extractors::{
-        extrude_db, BsoParam, CollectionParam, PreConditionHeader, PreConditionHeaderOpt,
-    },
+    extractors::{BsoParam, CollectionParam, PreConditionHeader, PreConditionHeaderOpt},
     middleware::SyncServerRequest,
     tags::Tags,
     DOCKER_FLOW_ENDPOINTS, X_LAST_MODIFIED,
 };
 
+use crate::db::transaction::DbTransactionPool;
 use actix_web::{
     dev::{Service, ServiceRequest, ServiceResponse, Transform},
     http::{header, StatusCode},
-    Error, HttpMessage, HttpResponse,
+    Error, FromRequest, HttpMessage, HttpResponse,
 };
-use futures::future::{self, Either, FutureExt, LocalBoxFuture, TryFutureExt};
+use futures::future::{self, FutureExt, LocalBoxFuture};
```

Comment:
Love the fact that we don't have to juggle the return values anymore and can kill `Either`